### ISSUE TITLE
panic when trying to add "" ""

### ISSUE THREAD

**Issue by tylert** [cbednarski/hostess#28]
Hostess 0.2.0 (https://github.com/cbednarski/hostess/releases/download/v0.2.0/hostess_linux_amd64) panics when given 2 empty strings...

```
$ sudo /usr/sbin/hostess add "" ""
panic: runtime error: index out of range

goroutine 1 [running]:
panic(0x5ecac0, 0xc820012080)
	/usr/local/go/src/runtime/panic.go:481 +0x3e6
github.com/cbednarski/hostess.Hostlist.Less(0xc82009c200, 0xa, 0x10, 0x2, 0x1, 0x0)
	/Users/cbednarski/go/src/github.com/cbednarski/hostess/hostlist.go:78 +0x559
github.com/cbednarski/hostess.(*Hostlist).Less(0xc82000ebe0, 0x2, 0x1, 0x0)
	<autogenerated>:2 +0xb5
sort.insertionSort(0x7f69450af578, 0xc82000ebe0, 0x0, 0xa)
	/usr/local/go/src/sort/sort.go:32 +0x6f
sort.quickSort(0x7f69450af578, 0xc82000ebe0, 0x0, 0xa, 0x8)
	/usr/local/go/src/sort/sort.go:214 +0x202
sort.Sort(0x7f69450af578, 0xc82000ebe0)
	/usr/local/go/src/sort/sort.go:229 +0x74
github.com/cbednarski/hostess.(*Hostlist).Sort(0xc82000a8d0)
	/Users/cbednarski/go/src/github.com/cbednarski/hostess/hostlist.go:140 +0x70
github.com/cbednarski/hostess.(*Hostlist).GetUniqueIPs(0xc82000a8d0, 0x0, 0x0, 0x0)
	/Users/cbednarski/go/src/github.com/cbednarski/hostess/hostlist.go:350 +0x4d
github.com/cbednarski/hostess.(*Hostlist).Format(0xc82000a8d0, 0x0, 0x0, 0x0)
	/Users/cbednarski/go/src/github.com/cbednarski/hostess/hostlist.go:381 +0xc0
github.com/cbednarski/hostess.(*Hostfile).Format(0xc82000a8c0, 0x0, 0x0, 0x0)
	/Users/cbednarski/go/src/github.com/cbednarski/hostess/hostfile.go:170 +0x3e
github.com/cbednarski/hostess.(*Hostfile).Save(0xc82000a8c0, 0x0, 0x0)
	/Users/cbednarski/go/src/github.com/cbednarski/hostess/hostfile.go:182 +0xbd
github.com/cbednarski/hostess.MaybeSaveHostFile(0xc8200963c0, 0xc82000a8c0)
	/Users/cbednarski/go/src/github.com/cbednarski/hostess/commands.go:69 +0x184
github.com/cbednarski/hostess.Add(0xc8200963c0)
	/Users/cbednarski/go/src/github.com/cbednarski/hostess/commands.go:113 +0x474
github.com/codegangsta/cli.Command.Run(0x62e4e0, 0x3, 0x0, 0x0, 0x0, 0x0, 0x0, 0x65d1a0, 0x1c, 0x0, ...)
	/Users/cbednarski/go/src/github.com/codegangsta/cli/command.go:179 +0x1399
github.com/codegangsta/cli.(*App).Run(0xc8200886e0, 0xc82000a140, 0x4, 0x4, 0x0, 0x0)
	/Users/cbednarski/go/src/github.com/codegangsta/cli/app.go:196 +0x137c
main.main()
	/Users/cbednarski/go/src/github.com/cbednarski/hostess/cmd/hostess/hostess.go:117 +0xa6c
```

It'll also quite happily enter bogus <nil> entries too like:
```
$ sudo /usr/sbin/hostess add "foo" ""
Added foo -> <nil> (On)
$ cat /etc/hosts
127.0.0.1 localhost
127.0.1.1 bamf
<nil> foo
::1 localhost ip6-localhost ip6-loopback
fe00:: ip6-localnet
ff00:: ip6-mcastprefix
ff02::1 ip6-allnodes
ff02::2 ip6-allrouters
```

(I recently switched from using https://github.com/hiteshjasani/nim-mungehosts to hostess and am still dealing with a few malformed callers.)

**Comment by cbednarski** [comment]
Thanks for the report! These are great fuzz cases.

I think these issues are fixed in master already, but I have not yet made a new release. If you have go installed you can `go get -u github.com/cbednarski/hostess`. I tried to repro using the version from master and I get:

> Failed to parse hosts entry: Unable to parse IP address ""

I'll get a new release out after I fix https://github.com/cbednarski/hostess/issues/27

**Comment by tylert** [comment]
:1st_place_medal:  No worries.  Figured I would report it since 0.2.0 was the current stable version.

Thanks again for yet another useful tool.

**Comment by tylert** [comment]
It would appear that the version string being reported by the pre-built 0.2.0 release binary is actually "0.1.0".

**Comment by cbednarski** [comment]
Oops! I'll get that fixed up in the next version. Thanks for the heads up!

**Comment by cbednarski** [comment]
Thanks for the report. I forget when I fixed this but the 0.3.0 release has the fix:

```
hostess add "" ""
Failed to parse hosts entry: Unable to parse IP address ""
```
